### PR TITLE
Fixes a null pointer that eventually happend

### PR DESCRIPTION
**Pull request by schmitch**
See https://github.com/playframework/playframework/issues/6151

### CODE REVIEW

File: netty-reactive-streams-http/src/main/java/com/typesafe/netty/http/HttpStreamsHandler.java [hunk 1]
```diff
@@ -291,7 +291,15 @@ public void run() {
...
     private void completeBody(final ChannelHandlerContext ctx) {
-        ctx.pipeline().remove(ctx.name() + "-body-subscriber");
+        // Netty 4.x will schedule a channelInactive event, which eventually
+        // will call a `DefaultChannelPipeline.destroy` which will remove all
+        // the handlers from the pipeline, so we need to check if the handler
+        // is still inside our pipeline
+        // See https://github.com/playframework/playframework/issues/6151
+        String channelName = ctx.name() + "-body-subscriber";
+        if (ctx.pipeline().get(channelName) != null) {
+            ctx.pipeline().remove(channelName);
```
Comment (jroper) [inline netty-reactive-streams-http/src/main/java/com/typesafe/netty/http/HttpStreamsHandler.java]: Does this actually fix the race condition?  I think it's possible that the channel could be removed between when we get it, and when we try to remove it.

**Comment by jroper** [comment]
I don't think this is the right solution - it's too broad.  We want to not remove the handler if the channel is already inactive.  If the handler is already removed for some other reason, then we want this to fail, as @gmethvin pointed out, doing that allowed us to detect a bug in Akka streams.

**Comment by jroper** [comment]
I've created #16 that only removes the handler if the channel is active.

**Comment by gmethvin** [comment]
@jroper Agreed. I was going to suggest a similar solution to #16. Thanks 👍

**Comment by schmitch** [comment]
After looking at #16 I agree, too. Way more clean.

**Pull request closed without merging**